software/ros_ws/src/perseus_sensors/src/m2m2_lidar/m2m2_lidar.cpp

```diff
@@ -548,6 +555,210 @@ void M2M2Lidar::_readSensorData()
+
+    if (input_points.empty())
+        return {};
+
+    // First normalize and sort points by angle
+    vector<tuple<float, float, bool>> points;
+    points.reserve(input_points.size());
+
+    // Normalize angles to [0, 2*pi] range
+    for (const auto& point : input_points)
+    {
+        float angle = get<0>(point);
+        /**
+         * @brief Normalize angle to [0, 2π] range
+         *
+         * @details This normalization is crucial for consistent LIDAR point processing:
+         *          - Ensures all angles are positive and less than 2*pi
+         *          - Handles wrap-around cases (e.g. -pi/2 becomes 3*pi / 2)
+         *
+         * The algorithm uses std::trunc to calculate the exact number of full rotations
+         * needed to avoid any potential precision loss from repeated floating-point operations.
+         */
+        const auto tau = 2 * std::numbers::pi;
+        angle -= std::trunc(angle / tau) * tau;
+        if (angle < 0)
+            angle += tau;
+        points.emplace_back(angle, get<1>(point), get<2>(point));
+    }
+
+    // Sort by angle
+    sort(points.begin(), points.end(),
+         [](const auto& a, const auto& b)
+         { return get<0>(a) < get<0>(b); });
+
+    vector<tuple<float, float, bool>> interpolated;
+    interpolated.reserve(INTERPOLATED_POINTS);
+
+    // Calculate angle increment
+    float angleIncrement = 2 * M_PI / INTERPOLATED_POINTS;
+
+    // Track index between iterations since points are sorted by angle
+    size_t sourcePointIndex = 0;
+
+    // For each desired interpolation point
+    for (size_t i = 0; i < INTERPOLATED_POINTS; ++i)
+    {
+        float targetAngle = i * angleIncrement;
+
+        // Find points that bracket our target angle
+        // C++20 approach using ranges to find next point
+        // 1. views::drop - Skip elements we've already processed
+        // 2. find_if - Search for first point with angle >= target
+        // 3. distance - Get index of found point
+
+        if (auto foundIterator = std::ranges::find_if(points | std::views::drop(sourcePointIndex),
+                                                      [targetAngle](const auto& point)
+                                                      {
+                                                          return std::get<0>(point) >= targetAngle;
+                                                      });
+            foundIterator != points.end())
+        {
+            sourcePointIndex = std::distance(points.begin(), foundIterator);
+        }
+        else
+        {
+            sourcePointIndex = points.size();
+        }
+
+        // Handle edge cases
+        if (sourcePointIndex == 0)
+        {
+            // Interpolate between last and first point
+            auto p1 = points.back();
+            auto p2 = points.front();
+
+            // Adjust angle for wrap-around
+            float a1 = get<0>(p1);
+            float a2 = get<0>(p2) + 2 * M_PI;
```

Comment:
```suggestion
            float a2 = get<0>(p2) + 2 * std::numbers::pi;
```
You'll need to `#include <numbers>` though.